pkg/flare/manifests.go

```diff
@@ -0,0 +1,206 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package flare
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/yaml"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var (
+	b64       = base64.StdEncoding
+	magicGzip = []byte{0x1f, 0x8b, 0x08}
+)
+
+const (
+	HELM_CHART_RELEASE_NAME       = "CHART_RELEASE_NAME"
+	HELM_CHART_RELEASE_NAMESPACE  = "DD_KUBE_RESOURCES_NAMESPACE"
+	HELM_AGENT_DAEMONSET          = "AGENT_DAEMONSET"
+	HELM_CLUSTER_AGENT_DEPLOYMENT = "CLUSTER_AGENT_DEPLOYMENT"
+)
+
+// chartUserValues is defined to unmarshall JSON data decoded from a Helm cart release into accessible fields
+type chartUserValues struct {
+	// user-defined values overriding the chart defaults
+	Config map[string]interface{} `json:"config,omitempty"`
+}
+
+// Retrieve a DaemonSet YAML from the API server for a given name and namespace, and returns the associated YAML manifest into a a byte array.
+// Its purpose is to retrieve the Datadog Agent DaemonSet manifest when building a Cluster Agent flare.
+func GetDaemonset(cl *apiserver.APIClient, name string, namespace string) ([]byte, error) {
+	var b bytes.Buffer
+
+	ds, err := cl.Cl.AppsV1().DaemonSets(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		log.Debugf("Can't retrieve DaemonSet %v from the API server: %s", name, err.Error())
+		return nil, err
+	}
+
+	dsjson, err := json.Marshal(ds)
+	ddyaml, err := yaml.JSONToYAML(dsjson)
+	fmt.Fprintln(&b, string(ddyaml))
+	return b.Bytes(), nil
```

Comment:
the function `yaml.JSONToYAML(dsjson)` already return `([]byte, error)` so no need to use a buffer here
you can simply 

```suggestion
	return `yaml.JSONToYAML(dsjson)
```